src/LinAlg/hiopVectorRajaPar.cpp

```diff
@@ -1514,5 +1514,30 @@ void hiopVectorRajaPar::copyFromDev() const
   resmgr.copy(data_host, data_dev_);
 }
 
+long long hiopVectorRajaPar::numOfNegVal_w_tol(const double &tol) const
+{  
+  double* data = data_dev_;
+  RAJA::ReduceSum<hiop_raja_reduce, int> sum(0);
+  RAJA::forall<hiop_raja_exec>( RAJA::RangeSegment(0, n_local_),
+    RAJA_LAMBDA(RAJA::Index_type i)
+    {
+      sum += (data_[i]<-tol);
```

Comment:
It looks like the kernel is using the member variable for host data (`data_`) instead of using the local copy of pointer to device data (`data`, a local copy of `data_dev_`). I think the underscore suffix should be removed (`data_` -> `data`).

Reply:
Sorry, my fault. It's a typo. I did create the point in line [1519](https://github.com/LLNL/hiop/blob/82a3dca1f995f191b29a416bea3a856606eb56b7/src/LinAlg/hiopVectorRajaPar.cpp#L1519) but forget to use it.